ext/opcache/jit/zend_jit_vm_helpers.c

```diff
@@ -922,6 +926,13 @@ zend_jit_trace_stop ZEND_FASTCALL zend_jit_trace_execute(zend_execute_data *ex,
 				} else if (count >= JIT_G(max_recursive_calls)) {
 					stop = ZEND_JIT_TRACE_STOP_DEEP_RECURSION;
 					break;
+				/* If the inline function is JITTed, backtrack this trace. */
+				} else if ( idx > JIT_G(jit_trace_inline_func_limit) && \
+					backtrack_link_to_inline_func < 0 && \
+					(ZEND_OP_TRACE_INFO(opline, offset)->trace_flags & ZEND_JIT_TRACE_JITED)) {
```

Comment:
I think you check ``idx`` improperly and in wrong place.
It should be checked in the next chunk, like

````
} else if (backtrack_link_to_inline_func > 0 && idx - baktrack_link_to_inline_func > JIT_G(jit_trace_inline_func_limit)) {
     ...
```

Reply:
we also don't use backslashes in multi-line ``if`` conditions.

Reply:
also, if we successfully inlined function into trace we should reset ``backtrack_link_to_inline_func``

Reply:
My idea is that when the trace is too long and idx exceeds the limit value, we check whether the inline function has been compiled at the start of the inline function. 
Do you mean to just judge the length of inline functions?

Reply:
> My idea is that when the trace is too long and idx exceeds the limit value

Then the name ``opcache.jit_trace_inline_func_limit`` doesn't reflect what you are doing and you might stop tracing directly without "backtracking".

I think your idea is less obvious and efficient.
We should be able to form quite long traces with many short getters and setters inlined.

> Do you mean to just judge the length of inline functions?

yes.

Reply:
Dear maintainer, Hope to get your reply~

Reply:
> ```
> 				} else if (ZEND_OP_TRACE_INFO(opline, offset)->trace_flags & ZEND_JIT_TRACE_JITED) {
> 					backtrack_link_to_inline_func = idx;
> 					link_to_inline_func_opline = opline;
> 				} 
> 				if (backtrack_link_to_inline_func > 0 && 
> 				idx - baktrack_link_to_inline_func > JIT_G(jit_trace_inline_func_limit)) {
> 					break;
> 				}
> ```

It hard to say without a full patch.
something similar, but you do ``break`` without setting ``end_opline`` and ``stop``. Do I miss something?

Reply:
Okay, Let me have an update.

Reply:
> > ```
> > 				} else if (ZEND_OP_TRACE_INFO(opline, offset)->trace_flags & ZEND_JIT_TRACE_JITED) {
> > 					backtrack_link_to_inline_func = idx;
> > 					link_to_inline_func_opline = opline;
> > 				} 
> > 				if (backtrack_link_to_inline_func > 0 && 
> > 				idx - baktrack_link_to_inline_func > JIT_G(jit_trace_inline_func_limit)) {
> > 					break;
> > 				}
> > ```
> 
> It hard to say without a full patch. something similar, but you do `break` without setting `end_opline` and `stop`. Do I miss something?

I have updated the patch and how about that ?

Reply:
> I have updated the patch and how about that ?

I'll able to review this only on Monday